pkg/apis/garden/v1beta1/helper/helpers.go

```diff
@@ -661,3 +663,55 @@ func ReadShootedSeed(shoot *gardenv1beta1.Shoot) (*ShootedSeed, error) {
 
 	return shootedSeed, nil
 }
+
+// GetK8SNetworks returns the Kubernetes network CIDRs for the Shoot cluster.
+func GetK8SNetworks(shoot *gardenv1beta1.Shoot) (gardencorev1alpha1.K8SNetworks, error) {
```

Comment:
Can you change https://github.com/gardener/gardener/blob/master/pkg/operation/shoot/shoot.go#L141 to use this function?